### ISSUE TITLE
ShaderParameterDescription.UsageMask

### ISSUE THREAD

**Issue by kburgoyne** [sharpdx/SharpDX#565]
The unmanaged source is 8 bits.  It appears those 8 bits are being duplicated in UsageMask in both 0..7 and 8..15.  This causes direct comparisons to the enum constants to fail, however masking against the enum constants will work because the mask strips off 8..15.

There is a bandaid.  Make sure any operations strip off bits 8..N.  If doing a direct comparison then mask against 0xFF or do an unchecked cast to byte before comparing.  Masking against the enum constants to see if any given flag is set should work without a bandaid.  Masking against the enums "All" constant might also make sense, although it would depend upon precisely the definition of "All" when it comes to potential new flag bits being added in the future.

**Comment by xoofx** [comment]
I did a fix 8dbe95716c432d6163ba08349ee765db57effdca to make sure the mask is 8 bytes.

It is not clear in your post what are the other problems. Could you elaborate? (Sorry, don't have time to dig into the details, so if you can make it clearer, that would help, thanks!)

**Comment by kburgoyne** [comment]
Great!  And sorry the rest was misleading to you.  The rest of my comment is really just suggesting workaround options for anyone else who encountered the problem before it was fixed.  That can remove some of the pressure to provide a fix if others don't get stuck on the problem.

I have to say I've managed to do some mildly advanced stuff with SharpDX and it has yet to fail me.  I've suspected it might be the problem a couple of times, but each time I've been wrong and it's been something else. :-)  Good work.

So far I've managed to use SharpDX to get a video file to play on the six sides of a rotating cube.  The next stunt I'm trying to work out is getting live webcam video to display on the six sides of that rotating cube.

Great work!

**Comment by kburgoyne** [comment]
Great!  And sorry the rest was misleading to you.  The rest of my comment is really just suggesting workaround options for anyone else who encountered the problem before it was fixed.  That can remove some of the pressure to provide a fix if others don't get stuck on the problem.

 

I have to say I've managed to do some mildly advanced stuff with SharpDX and it has yet to fail me.  I've suspected it might be the problem a couple of times, but each time I've been wrong and it's been something else. :-)  Good work.

 

So far I've managed to use SharpDX to get a video file to play on the six sides of a rotating cube.  The next stunt I'm trying to work out is getting live webcam video to display on the six sides of that rotating cube.

 

Great work!
